lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -74,21 +74,42 @@ def rh_repos(server_version)
 
       def ansible_repo(server_version)
         if server_version >= version('6.8')
-          "rhel-#{el_major_version}-server-ansible-2.9-rpms"
+          ansible_version = '2.9'
         elsif server_version >= version('6.6')
-          "rhel-#{el_major_version}-server-ansible-2.8-rpms"
+          ansible_version = '2.8'
         elsif server_version >= version('6.4')
-          "rhel-#{el_major_version}-server-ansible-2.6-rpms"
+          ansible_version = '2.6'
+        end
+
+        if el7?
+          "rhel-#{el_major_version}-server-ansible-#{ansible_version}-rpms"
+        else
+          "ansible-#{ansible_version}-for-rhel-#{el_major_version}-x86_64-rpms"
         end
       end
 
+      def use_beta_repos?
+        ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
+      end
+
       def product_specific_repos(full_version)
         repos = []
-        repos << if ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
-                   "rhel-server-#{el_major_version}-#{package_name}-6-beta-rpms"
-                 else
-                   "rhel-#{el_major_version}-server-#{package_name}-#{full_version}-rpms"
-                 end
+
+        if el7?
+          if use_beta_repos?
+            repos << "rhel-server-#{el_major_version}-#{package_name}-6-beta-rpms"
+          else
+            repos << "rhel-#{el_major_version}-server-#{package_name}-#{full_version}-rpms"
+          end
+        else
+          if use_beta_repos?
+            # FIXME: this is for sure wrong
+            repos << "satellite-#{package_name}-for-rhel-#{el_major_version}-x86_64-rpms"
+          else
+            repos << "satellite-#{package_name}-for-rhel-#{el_major_version}-x86_64-rpms"
```

Comment:
What would package_name be here? I think this would match capsule/maintenance/client pattern but not the primary repository.

Reply:
Yeah, its as per the feature, Satellite/Capsule. So will be resolved to `satellite`/`capsule`.